docs/tutorials/getting-started-with-csharp/microservices.md

```diff
@@ -0,0 +1,443 @@
+of this tutorial, you'll learn:
+
+* How to generate an ASP.NET Core application using Yeoman
+* How to create a development Docker environment
+* How to build a Docker image based on an existing image.
+* How to deploy your service into a Docker container.
+
+Along the way, you'll also see some C# language features:
+
+* How to convert C# objects into JSON payloads.
+* Build immutable Data Transfer Objects
+* Process incoming HTTP Requests and generate the HTTP Response
+* Work with nullable value types
+
+## Why Docker?
+
+Docker makes it easy to create standard machine images to
+host your services in a data center, or the public cloud. Docker
+enables you to configure the image, and replicate it as needed to
+scale the installation of your application.
+
+All the code in this tutorial will work in any .NET Core environment.
+The additional tasks for a Docker installation will work for an ASP.NET
+Core application. 
+
+# Prerequisites
+You’ll need to setup your machine to run .NET core. Instructions are
+[here](http://dotnet.github.io/getting-started/). You can run this
+application on Windows, Ubuntu Linux, OS X or in a Docker container. 
+You’ll need to install your favorite code editor. The descriptions below
+use [Visual Studio Code](https://code.visualstudio.com/) which is an open
+source, cross platform editor. However, you can use whatever tools you are
+comfortable with.
+
+You'll also need to install the Docker engine. Instructions for that are
+located [here](https://docs.docker.com/engine/installation/). Docker can
+be installed in many Linux distributions, Mac OS, or Windows. The page
+referenced above contains links to each of the available installations.
+
+You'll also need to install a number of command line tools that support
+asp.net core development. The command line templates use yeoman, bower,
+grunt, and gulp. You may already have many of these tools, but if not,
+run the following command in your favorite shell:
+
+`npm install -g yo bower grunt-cli gulp`
+
+This instructs the node package manager (npm) to install the needed tools.
+The '-g' option indicates that it is a global install, and those tools are
+available system wide. (A local install scopes the package to a single
+project). Once you've installed those core tools, you need to install
+the yeoman asp.net template generators:
+
+Then, you can install ASP.NET template generator:
+
+`npm install -g generator-aspnet`
+
+# Create the Application
+
+Now that you've installed all the tools, create a new asp.net core
+application. To use the command line generator, execute the following
+yeoman command in your favorite shell:
+
+`yo aspnet`
+
+This command prompts you to select what type of application you want to
+create. For this microservice, you want the simplest, most light weight
+web application possible, so select 'Empty Application'. The template
+will prompt you for a name. Select 'WeatherMicroservice'. 
```

Comment:
Maybe Type would be better than select, unless you're actually selecting from a list or something